app/models/ems_refresh/save_inventory.rb

```diff
@@ -309,4 +309,31 @@ def save_snapshots_inventory(vm, hashes)
   def save_event_logs_inventory(os, hashes)
     save_inventory_multi(os.event_logs, hashes, :use_association, [:uid])
   end
+
+  def save_new_target(target_hash)
+    vm_hash = target_hash[:vm]
+    existing_vm = VmOrTemplate.find_by(:ems_ref => vm_hash[:ems_ref])
```

Comment:
As we get to creating new targets that aren't VMs we'll want to check for `target_hash[:vm].nil?`